src/locked.rs

```diff
@@ -23,14 +30,13 @@ impl LockedWallet {
 
     pub fn unlock(&self, key: &[u8]) -> Result<UnlockedWallet, Error> {
         let mut sha3 = Sha3_256::new();
-        sha3.input(key);
-        let pass = sha3.result();
-
-        let x_cha_cha = SymmetricEncryptor::<XChaCha20Poly1305>::new_with_key(pass)
-            .map_err(|e| Error::AeadCryptoError(e))?;
+        sha3.update(key);
+        let pass = sha3.finalize();
 
-        let dec = x_cha_cha
-            .decrypt_easy(self.id.as_bytes(), &self.ciphertext)
+        let cha_cha = XChaCha20Poly1305::new(&pass);
+        let nonce = XNonce::from_slice(self.id.as_bytes());
```

Comment:
we should avoid use the id as nonce for a few reasons:
- restricts us to 24 byte IDs (although we could always resize anything to 24 bytes)
- forces repeated use of the same nonce (generally a bad idea to use the same nonce more than once)
Generally for a purpose like this, the nonce is randomly generated and prepended or appended to the cipertext, and they can then be separated and plugged in as arguements (like when the key is separated from the ciphertext in `seal_box`, check out the [impl in ursa of `encrypt_easy` and `decrypt_easy`](https://github.com/hyperledger/ursa/blob/master/libursa/src/encryption/symm/mod.rs#L237))

Reply:
appending random to message now